{"timestamp":"2025-08-21T08:06:55.882438","level":"info","event":"DAG bundles loaded: dags-folder, example_dags","logger":"airflow.dag_processing.bundles.manager.DagBundlesManager"}
{"timestamp":"2025-08-21T08:06:55.883479","level":"info","event":"Filling up the DagBag from /opt/airflow/dags/fetchin_cleaning.py","logger":"airflow.models.dagbag.DagBag"}
{"timestamp":"2025-08-21T08:06:55.972006Z","level":"info","event":"Task instance is in running state","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:55.972622Z","level":"info","event":" Previous state of the Task instance: TaskInstanceState.QUEUED","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:55.974955Z","level":"info","event":"Current task name:clean_data","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:55.975375Z","level":"info","event":"Dag name:weather_to_snowflake_pipeline","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.812543Z","level":"error","event":"Traceback (most recent call last):","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.813120Z","level":"error","event":"  File \"/opt/airflow/dags/cleaning_data.py\", line 167, in <module>","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.816148Z","level":"error","event":"    data_cleaning()","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.816581Z","level":"error","event":"  File \"/opt/airflow/dags/cleaning_data.py\", line 129, in data_cleaning","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.819031Z","level":"error","event":"    pollutants_fact_new = pollutants_fact_new.merge(date_dim[['date_sk','date_time']], left_on='pollution_components_dt', right_on='date_time', how='left')","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.819484Z","level":"error","event":"                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.819816Z","level":"error","event":"  File \"/home/airflow/.local/lib/python3.12/site-packages/pandas/core/frame.py\", line 10487, in merge","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.820720Z","level":"error","event":"    return merge(","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.821078Z","level":"error","event":"           ^^^^^^","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.821412Z","level":"error","event":"  File \"/home/airflow/.local/lib/python3.12/site-packages/pandas/core/reshape/merge.py\", line 169, in merge","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.821801Z","level":"error","event":"    op = _MergeOperation(","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.822116Z","level":"error","event":"         ^^^^^^^^^^^^^^^^","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.822400Z","level":"error","event":"  File \"/home/airflow/.local/lib/python3.12/site-packages/pandas/core/reshape/merge.py\", line 791, in __init__","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.822688Z","level":"error","event":"    ) = self._get_merge_keys()","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.823008Z","level":"error","event":"        ^^^^^^^^^^^^^^^^^^^^^^","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.823296Z","level":"error","event":"  File \"/home/airflow/.local/lib/python3.12/site-packages/pandas/core/reshape/merge.py\", line 1287, in _get_merge_keys","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.823584Z","level":"error","event":"    left_keys.append(left._get_label_or_level_values(lk))","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.823867Z","level":"error","event":"                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.824128Z","level":"error","event":"  File \"/home/airflow/.local/lib/python3.12/site-packages/pandas/core/generic.py\", line 1844, in _get_label_or_level_values","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.824471Z","level":"error","event":"    raise KeyError(key)","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.824773Z","level":"error","event":"KeyError: 'pollution_components_dt'","chan":"stderr","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.961776","level":"error","event":"Task failed with exception","logger":"task","error_detail":[{"exc_type":"CalledProcessError","exc_value":"Command '['python', 'dags/cleaning_data.py']' returned non-zero exit status 1.","exc_notes":[],"syntax_error":null,"is_cause":false,"frames":[{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":890,"name":"run"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":1177,"name":"_execute_task"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/bases/operator.py","lineno":397,"name":"wrapper"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/standard/operators/python.py","lineno":217,"name":"execute"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/standard/operators/python.py","lineno":240,"name":"execute_callable"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/callback_runner.py","lineno":81,"name":"run"},{"filename":"/opt/airflow/dags/fetchin_cleaning.py","lineno":16,"name":"clean_data"},{"filename":"/usr/local/lib/python3.12/subprocess.py","lineno":571,"name":"run"}],"is_group":false,"exceptions":[]}]}
{"timestamp":"2025-08-21T08:06:57.964189Z","level":"info","event":"Task instance in failure state","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.964739Z","level":"info","event":"Task start","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.971187Z","level":"info","event":"Task:<Task(PythonOperator): clean_data>","chan":"stdout","logger":"task"}
{"timestamp":"2025-08-21T08:06:57.971836Z","level":"info","event":"Failure caused by Command '['python', 'dags/cleaning_data.py']' returned non-zero exit status 1.","chan":"stdout","logger":"task"}
